src/types.ts

```diff
@@ -1,32 +1,23 @@
 /* Type Definitions */
 
-export type Collection = {
-  id: string;
-  createdAt: Date;
-  updatedAt: Date;
-  embeddingSource: string;
-  indexingStatus: 'indexing' | 'ready';
-  source: string;
-  metadata: any;
+export type ObjectProperties = {
+  [key: string]: any;
 };
 
-export type Group = {
+export type Object = {
   id: string;
+  parentId?: string;
   createdAt: Date;
   updatedAt: Date;
-  indexingStatus: 'indexing' | 'ready';
-  parentId?: string;
-  collectionId?: string;
-  properties: any; // Near-arbritrary.
-  related?: Group[]; // Only used if ?related=N is passed to getGroup.
-  atoms?: number; // Only if ?atoms=true is passed to getGroup.
+  properties: ObjectProperties;
+  indexingStatus: 'indexed' | 'ready';
 } & (
   | {
-      kind: 'collection';
+      type: 'collection';
       metadata: {};
     }
   | {
-      kind: 'text';
+      type: 'text';
       metadata: {
         text: string;
       };
```

Comment:
Can we extract the metadata objects into separate types? Then this type becomes:

kind: string;
metadata: CollectionMetadata | TextMetadata | ...

This makes typecasting much easier, rather than the dumb `object.metadata as {text: string};` nonsense that we currently have to do in the /v2/ API.

Reply:
I did this but it didn't make type errors. Can still use it for casting tho so yes